### PR TITLE
Map.ofAll(Stream, Mapper) #1341

### PR DESCRIPTION
**Pull request by ruslansennov**

### CODE REVIEW

File: javaslang/src/test/java/javaslang/collection/AbstractMapTest.java [hunk 1]
```diff
@@ -273,6 +278,20 @@ public void shouldBeTheSame() {
         return new java.util.AbstractMap.SimpleEntry<>(key, value);
     }
 
+    @Test
+    public void shouldConstructFromJavaStream() {
+        final java.util.stream.Stream<Integer> javaStream = java.util.stream.Stream.of(1, 2, 3);
+        final Map<String, Integer> map = mapOf(javaStream, String::valueOf, Function.identity());
+        assertThat(map).isEqualTo(this.<String, Integer> emptyMap().put("1", 1).put("2", 2).put("3", 3));
```
Comment (sputnikci) [inline javaslang/src/test/java/javaslang/collection/AbstractMapTest.java]: \[Checkstyle\] ERROR: '>' is followed by whitespace\.

File: javaslang/src/test/java/javaslang/collection/AbstractMapTest.java [hunk 2]
```diff
@@ -273,6 +278,20 @@ public void shouldBeTheSame() {
...
+    }
+
+    @Test
+    public void shouldConstructFromJavaStreamEntries() {
+        final java.util.stream.Stream<Integer> javaStream = java.util.stream.Stream.of(1, 2, 3);
+        final Map<String, Integer> map = mapOf(javaStream, i -> Tuple.of(String.valueOf(i), i));
+        assertThat(map).isEqualTo(this.<String, Integer> emptyMap().put("1", 1).put("2", 2).put("3", 3));
```
Comment (sputnikci) [inline javaslang/src/test/java/javaslang/collection/AbstractMapTest.java]: \[Checkstyle\] ERROR: '>' is followed by whitespace\.

File: javaslang/src/main/java/javaslang/collection/Maps.java [hunk 3]
```diff
@@ -51,6 +51,24 @@ private Maps() {
...
+    }
+
+    @SuppressWarnings("unchecked")
+    static <T, K, V, M extends Map<K, V>> M ofStream(M map, java.util.stream.Stream<? extends T> stream,
+                                                     Function<? super T, Tuple2<? extends K, ? extends V>> entryMapper) {
+        Objects.requireNonNull(stream, "stream is null");
+        Objects.requireNonNull(entryMapper, "entryMapper is null");
+        return Stream.ofAll(stream).foldLeft(map, (m, el) -> (M) m.put(entryMapper.apply(el)));
+    }
+
```
Comment (danieldietrich) [inline javaslang/src/main/java/javaslang/collection/Maps.java]: please move the `ofStream` methods behind `merge` (I sorted the methods alphabetically)

Comment (ruslansennov) [inline javaslang/src/main/java/javaslang/collection/Maps.java]: done

**Pull request merged**